rust/noosphere-core/src/view/sphere.rs

```diff
@@ -791,6 +796,58 @@ impl<S: BlockStore> Sphere<S> {
         }
     }
 
+    /// Verify that a given authorization is a valid with regards to operating
+    /// on this [Sphere]; it is issued by the sphere (or appropriately
+    /// delegated), and it has not been revoked.
+    pub async fn verify_authorization(&self, authorization: &Authorization) -> Result<()> {
+        let proof_chain = authorization
+            .as_proof_chain(&UcanStore(self.store.clone()))
+            .await?;
+
+        let authority = self.get_authority().await?;
+        let delegations = authority.get_delegations().await?;
+        let revocations = authority.get_revocations().await?;
+        let sphere_identity = self.get_identity().await?;
+        let sphere_credential = sphere_identity.to_credential()?;
+
+        let mut remaining_links = vec![&proof_chain];
+
+        while let Some(chain_link) = remaining_links.pop() {
+            let link = Link::from(Cid::try_from(chain_link.ucan())?);
+
+            if delegations.get(&link).await?.is_none() {
+                return Err(anyhow!(
+                    "Authorization {} not found in sphere authority",
+                    link
+                ));
+            }
+
+            if let Some(revocation) = revocations.get(&link).await? {
+                // NOTE: The implication here is that only the sphere itself can
+                // issue revocations The follow-on implicationis that a user
+                // must provide the sphere mnemonic in order to issue a
+                // revocation
+                if revocation.iss != sphere_identity {
```

Comment:
Per the comment, the sphere root key always _issues_ the revocation. The sphere cannot revoke its own authority; it is the alpha and the omega, it is self-sovereign, what have you. And, as the root of all authority it may revoke any other authority delegated from it or another of its delegatees (direct or transitive).